### PR TITLE
feat(semseg): allow model customization

### PR DESCRIPTION
**Pull request by borisdayma**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
*not applicable*
- [ ] Did you write any new necessary tests?  
*not applicable*
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Allow more customization to the semantic segmentation example:

* test set has been replace with a validation set to measure model performance
* both training and validation losses are logged
* unet allows for more customization: number of layers and number of features
* masks have been updated to be compatible with the most recent version of the dataset which contains more labels (additional labels are just ignored)
* script allows for other arguments: distributed backend, 16 bit, number of layers and features for unet as well as type of upsampling, gradient accumulation, number of epochs, logging on W&B

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by borisdayma** [comment]
Note: this example requires #1311

**Comment by borisdayma** [comment]
The build error related to "examples" mentions problem of relative import which was already there before. Not sure if you want to change it. I think it's clean this way.

**Comment by Borda** [comment]
@borisdayma can we finish this one when #1311 is done... :]

**Comment by borisdayma** [comment]
I think this PR can now be closed in favor of #1371.
